libraries/AP_RangeFinder/AP_RangeFinder_uLanding.cpp

```diff
@@ -55,25 +57,31 @@ bool AP_RangeFinder_uLanding::get_reading(uint16_t &reading_cm)
     // read any available lines from the uLanding
     float sum = 0;
     uint16_t count = 0;
-    uint8_t  index = 0;
+    bool hdr_found = false;
 
     int16_t nbytes = uart->available();
     while (nbytes-- > 0) {
         uint8_t c = uart->read();
-        // ok, we have located start byte
-        if (c == 72 && index == 0) {
+        
+        if (c == ULANDING_HDR && !hdr_found) {
+            // located header byte
             linebuf_len = 0;
-            index       = 1;
+            hdr_found   = true;
         }
-        // now it is ready to decode index information
-        if (index == 1) {
-            linebuf[linebuf_len] = c;
-            linebuf_len ++;
-            if (linebuf_len == 3) {
-                index = 0;
-                sum += (linebuf[2]&0x7F) *128 + (linebuf[1]&0x7F);
+        // decode index information
+        if (hdr_found) {
+            linebuf[linebuf_len++] = c;
+
+            // process linebuf once we've collected six bytes of data
+            if (linebuf_len == 6) {
```

Comment:
A more elegant and safer way of stating this is:
```
if (linebuf_len < 6) {
  linebuf[linebuf_len++] = c;
} else {
  // evaluate checksum
  ...
  linebuf_len=0;
}
```
Even though the existing code works, this will make it obvious to the reader/compiler that it can never exceed the array limits.

Reply:
@magicrub I need it to fill the last index in linebuf with the current byte and go straight to evaluating the checksum. Would the below example be ok with you? Same idea, just slightly different execution.
'''
if (hdr_found) {
    linebuf[linebuf_len++] = c;

    if (linebuf_len < 6) {
        continue;
    } else {
        // evaluate checksum
        ...
        linebuf_len = 0;
    }
}
'''

Reply:
ok, if we're getting into the details this really what it needs to verify. Basically, just never allow it to accidentily have a memory leak. ever.
```
if (linebuf_len < sizeof(linebuf)) { 
  linebuf[linebuf_len++] = c;
  ...
} else {
  // Glad we checked because we almost had a memory leak!
  linebuf_len = 0;
  hdr_found = false;
}
```